### ISSUE TITLE
All GUI apps suddenly vanished

### ISSUE THREAD

**Issue by shebiki** [microsoft/wslg#431]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.22000.168]
Your Distribution version: Ubuntu 20.04
Your WSLg version: 1.0.26
```

# Steps to reproduce

Not sure what triggered this, was happily developing away when all my Linux GUI apps closed.

**WSL logs**: 

* [pulseaudio.log](https://github.com/microsoft/wslg/files/7085345/pulseaudio.log)
* [weston.log](https://github.com/microsoft/wslg/files/7085346/weston.log)
* `versions.txt`
```
WSLg ( x86_64 ): 1.0.26+Branch.main.Sha.26ce2c09b86442f3c7f4f6462f770ed2afa76a25
Mariner: VERSION="1.0.20210224"
FreeRDP: b05321cd4e6a862aef76163a69db4e1910245736
weston: 46756d0e77e5c01b5995fbbee6f3ab0db9b30612
pulseaudio: 2f0f0b8c3872780f15e275fc12899f4564f01bd5
mesa:
```

**Comment by shebiki** [comment]
[core.weston.zip](https://github.com/microsoft/wslg/files/7085369/core.weston.zip)

**Comment by shebiki** [comment]
Just happened again, updated logs attached. Was using a combination of mouse and keyboard when weston crashed.

[core.weston.zip](https://github.com/microsoft/wslg/files/7085607/core.weston.zip)
[pulseaudio.log](https://github.com/microsoft/wslg/files/7085608/pulseaudio.log)
[stderr.log](https://github.com/microsoft/wslg/files/7085609/stderr.log)
[versions.txt](https://github.com/microsoft/wslg/files/7085610/versions.txt)
[weston.log](https://github.com/microsoft/wslg/files/7085611/weston.log)

**Comment by hideyukn88** [comment]
@shebiki, thanks for reporting the issue and dumps, this issue should be fixed by https://github.com/microsoft/weston-mirror/pull/39, and the fix will be included in 1.0.28, thanks!
